src/monster.cpp

```diff
@@ -724,8 +726,13 @@ void Monster::onThink(uint32_t interval)
 	}
 
 	if (!isInSpawnRange(position)) {
-		g_game.internalTeleport(this, masterPos);
-		setIdle(true);
+		if (g_config.getBoolean(ConfigManager::REMOVE_ON_DESPAWN)) {
+			g_game.addMagicEffect(this->getPosition(), CONST_ME_POFF);
+			g_game.removeCreature(this, false);
+		} else {
+			g_game.internalTeleport(this, masterPos);
```

Comment:
I think the magic effect should be added no matter the config. The removal is transparent from the player, so visually it should be the same.